wall_e/src/resources/cogs/ban.py

```diff
@@ -0,0 +1,419 @@
+
+        # confirm at least 1 @ mention of user to ban
+        if len(ctx.message.mentions) < 1:
+            logger.info("[Ban ban()] No user were @ mentioned in the args")
+            e_obj = await em(ctx=ctx, title="Invalid Arguments",
+                             content=[("Error", "Please @ mention the user to ban"),
+                                      ("Command Usage", "`.ban @user [<# of days to purge messages>] [<reason>]`"),
+                                      ("Example Usage", "`.ban @user1 2 they're being weird`")],
+                             colour=self.error_colour,
+                             footer="Command Error")
+            if e_obj:
+                await ctx.send(embed=e_obj)
+            return
+
+        logger.info(f"[Ban ban()] User to ban: {user}")
+
+        # construct reason
+        reason = ' '.join(reason)
+        reason = reason if reason else "No Reason Given."
+        logger.info(f"[Ban ban()] Ban reason{reason}")
+
+        # ban
+        dm = True
+        # for user in users_to_ban:
+        ban = BanRecords(
+                         username=user.name + '#' + user.discriminator,
+                         user_id=user.id,
+                         mod=ctx.author.name+'#'+ctx.author.discriminator,
+                         mod_id=ctx.author.id,
+                         reason=reason
+                         )
+
+        logger.info(f"[Ban ban()] Banning {ban.username} with id {ban.user_id}")
+
+        # add to ban_list
+        self.ban_list.append(ban.user_id)
+
+        # dm banned user
+        e_obj = discord.Embed(title="Ban Notification",
+                              description="You have been **PERMANENTLY BANNED** from " +
+                              f"**{self.bot.guilds[0].name.upper()}**",
+                              color=discord.Color.red(),
+                              timestamp=datetime.datetime.now(pytz.utc)
+                              )
+        e_obj.add_field(name='Reason',
+                        value=f"```{reason}```\n" +
+                        "**Please refrain from this kind of behaviour in the future. Thank you.**"
+                        )
+
+        e_obj.set_footer(icon_url=self.bot.guilds[0].icon_url, text=self.bot.guilds[0])
+        try:
+            await user.send(embed=e_obj)
+            logger.info("[Ban ban()] User notified via dm of their ban")
+        except (discord.HTTPException, discord.Forbidden, discord.InvalidArgument):
+            dm = False
+            logger.info("[Ban ban()] Notification dm to user failed due to user preferences")
+
+        # kick
+        await user.kick(reason=reason)
+        dt = datetime.datetime.now(pytz.utc)
+        ban.ban_date = dt.timestamp()
+
+        logger.info(f"[Ban ban()] User kicked from guiled at {dt}.")
+
+        # begin purging messages from user in the last purge_window_days
+        await self.purge_messages(ctx, user, purge_window_days)
```

Comment:
and you dont verify that the purge_window_days validated in `purge_messages` are correct even if the command gets aborted